File: app/src/main/java/com/app/turingrobot/ui/core/BaseFragment.java
package com.app.turingrobot.ui.core;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import com.app.turingrobot.core.App;
import com.app.turingrobot.core.RobotService;
import com.app.turingrobot.core.dagger.component.DaggerFragmentComponent;
import com.app.turingrobot.core.dagger.module.FModule;
import com.app.turingrobot.helper.SpfHelper;

import javax.inject.Inject;

/**
 * Created by Alpha on 2016/3/26 21:55.
 */
public class BaseFragment extends Fragment {

    protected App mApp = App.get();

    @Inject
    protected RobotService apiService;

    @Inject
    protected SpfHelper mSpf;

    protected AppCompatActivity activity;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.activity = (AppCompatActivity) context;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        inject();
    }

    private void inject() {
        DaggerFragmentComponent.builder()
                .fModule(new FModule())
                .appComponent(mApp.getComponent())
                .build().inject(this);
    }

}

File: app/src/main/java/com/app/turingrobot/ui/core/BaseFragment.kt
package com.app.turingrobot.ui.core

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity

import com.app.turingrobot.app.App
import com.app.turingrobot.core.RobotService
import com.app.turingrobot.helper.SpfHelper

import javax.inject.Inject

/**
 * Created by Alpha on 2016/3/26 21:55.
 */
open class BaseFragment : Fragment() {

    @Inject
    lateinit var apiService: RobotService

    @Inject
    lateinit var mSpf: SpfHelper

    protected var activity: AppCompatActivity? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        this.activity = context as AppCompatActivity
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        inject()
    }

    private fun inject() {
        App.component.inject(this)
    }

}
